analyzer/src/steps/typing/structure.rs

```diff
@@ -0,0 +1,446 @@
+use std::collections::HashMap;
+
+use ast::r#struct::{FieldAccess, StructDeclaration};
+use ast::variable::Assign;
+use context::source::{SourceSegment, SourceSegmentHolder};
+
+use crate::diagnostic::{Diagnostic, DiagnosticID, Observation};
+use crate::reef::ReefId;
+use crate::relations::{LocalId, SymbolRef};
+use crate::steps::typing::bounds::{apply_bounds, TypesBounds};
+use crate::steps::typing::coercion::resolve_type_annotation;
+use crate::steps::typing::exploration::{Exploration, Links};
+use crate::steps::typing::{ascribe_types, type_rhs, ExpressionValue, TypingState};
+use crate::types::engine::{Chunk, StructureId};
+use crate::types::hir::{ExprKind, TypedExpr};
+use crate::types::ty::{Field, FunctionDesc, Type, TypeRef};
+use crate::types::{hir, ERROR, UNIT};
+
+pub(super) fn declare_structure(
+    decl: &StructDeclaration,
+    exploration: &mut Exploration,
+    parent_links: Links,
```

Comment:
Functions use the `function_links`, is there any particular reason to do declare all things using the parent environment? The local and the type should be in the parent environment, and the parameters in the definition environment. Interchanging them in this function is very strange.